src/System.Memory/tests/ParsersAndFormatters/Parser/TestData.Parser.Date.cs

```diff
@@ -51,8 +77,33 @@ public static IEnumerable<ParserTestData<DateTimeOffset>> DateTimeOffsetParserTe
         {
             get
             {
-                // Wrong day of week.
-                yield return new ParserTestData<DateTimeOffset>("Thu, 13 Jan 2017 03:45:32 GMT", default, 'R', expectedSuccess: false);
+				// Format J tests for DateTimeOffset parsing.
+				yield return new ParserTestData<DateTimeOffset>("0997-07-16", DateTimeOffset.Parse("0997-07-16"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16", DateTimeOffset.Parse("1997-07-16"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16T19:20", DateTimeOffset.Parse("1997-07-16T19:20"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16T19:20:30", DateTimeOffset.Parse("1997-07-16T19:20:30"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16T19:20:30.45", DateTimeOffset.Parse("1997-07-16T19:20:30.45"), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16T19:20:30.4555555", DateTimeOffset.Parse("1997-07-16T19:20:30.4555555"), 'J', expectedSuccess: true);
+
+				// Test fraction rounding.
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16T19:20:30.45555554", DateTimeOffset.Parse("1997-07-16T19:20:30.45555554"), 'J', expectedSuccess: true);
+				// DateTimeOffset.Parse will round up to 7dp in this case, but we expect parser to truncate.
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16T19:20:30.45555555", DateTimeOffset.Parse("1997-07-16T19:20:30.4555555"), 'J', expectedSuccess: true);
+
+				// Test with timezone designator.
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16T19:20:30.4555555Z", DateTimeOffset.ParseExact("1997-07-16T19:20:30.4555555Z", "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16T19:20:30.4555555+01:00", DateTimeOffset.ParseExact("1997-07-16T19:20:30.4555555+01:00", "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind), 'J', expectedSuccess: true);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16T19:20:30.4555555-01:00", DateTimeOffset.ParseExact("1997-07-16T19:20:30.4555555-01:00", "O", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind), 'J', expectedSuccess: true);
+
+				// Invalid strings.
+				yield return new ParserTestData<DateTimeOffset>("997-07-16", default, 'J', expectedSuccess: false);
+				yield return new ParserTestData<DateTimeOffset>("1997-07", default, 'J', expectedSuccess: false);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-6", default, 'J', expectedSuccess: false);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16Z", default, 'J', expectedSuccess: false);
+				yield return new ParserTestData<DateTimeOffset>("1997-07-16+01:00", default, 'J', expectedSuccess: false);
```

Comment:
If you are able to run it, what does the code coverage look like: https://github.com/dotnet/corefx/blob/master/Documentation/building/code-coverage.md#code-coverage-with-systemprivatecorelib-code